python/heterocl/devices.py

```diff
@@ -287,17 +312,45 @@ def __init__(self, name, devs, host, xcel, tool):
 
         # attach supported memory modules
         if xcel.vendor == "xilinx" and "xcvu19p" in xcel.model:
-            self.host.storage["hbm"]   = HBM()
-            self.host.storage["plram"] = PLRAM()
-            self.host.storage["ssd"]   = SSD()
-            self.xcel.storage["hbm"]   = HBM()
-            self.xcel.storage["plram"] = PLRAM()
-            self.xcel.storage["ssd"]   = SSD()
+            self.host.storage["HBM"]   = HBM()
+            self.host.storage["PLRAM"] = PLRAM()
+            self.xcel.storage["HBM"]   = HBM()
+            self.xcel.storage["PLRAM"] = PLRAM()
+
+        # attach on-device memory devices
+        self.xcel.storage["URAM"] = URAM()
+        self.xcel.storage["BRAM"] = BRAM()
+        self.xcel.storage["LUTRAM"] = LUTRAM()
```

Comment:
Can you show how this should be used?

Reply:
We have some test cases to test data movement to these targets. https://github.com/Hecmay/heterocl/blob/merge/tests/test_schedule_stream.py#L280

We just use it as the target media for .to API